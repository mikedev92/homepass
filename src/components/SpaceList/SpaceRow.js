// @flow
import React from 'react';
import { Row, Left, View, Text, ListItem, Body, Thumbnail } from 'native-base';
import { withNavigation } from 'react-navigation';
import gql from 'graphql-tag';
import moment from 'moment';
import { compose, setStatic } from 'recompose';
import { get } from 'lodash';

type Props = {
	space: Space,
	navigation: Object,
	last: Boolean
};
function SpaceRow({ space, navigation, last }: Props) {
	const { id, title, subtitle, imageUrl, nextEvent } = space;
	const memberEdges: SpaceMemberEdge[] = get(space, 'members.edges') || [];
	return (
		<ListItem thumbnail onPress={() => navigation.navigate('Space', { id, title })} last={last}>
			<Left>
				<Thumbnail square style={{ borderRadius: 5 }} source={{ uri: imageUrl }} />
			</Left>
			<Body>
				<Text>
					{title}
					{'\n'}
					<Text note>
						{subtitle}
					</Text>
					{'\n'}
					{nextEvent &&
						<Text note>
							{nextEvent.name} {moment(nextEvent.startDate).fromNow()}
						</Text>}
				</Text>
				<Row>
					<View style={{ flex: 1 }} />
					{memberEdges.map(member => member && member.node && member.node.user && member.node.user.imageUrl && <Thumbnail xSmall key={member.node.id} source={{ uri: member.node.user.imageUrl }} style={{ marginHorizontal: 2 }} />)}
				</Row>
			</Body>
		</ListItem>
	);
}

export default compose(
	setStatic('fragments', {
		space: gql`
			fragment SpaceRowSpace on Space {
				id
				title
				imageUrl
				subtitle
				nextEvent {
					startDate
					endDate
					name
				}
				members {
					edges {
						node {
							id
							user {
								id
								imageUrl
							}
						}
					}
				}
			}
		`
	}),
	withNavigation
)(SpaceRow);
