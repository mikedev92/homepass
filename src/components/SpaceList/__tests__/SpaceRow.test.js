import React from 'react';
import { shallow } from 'enzyme';
import SpaceRow from '../SpaceRow';

describe('<SpaceRow />', () => {
  it('matches snapshot', () => {
    const testProps = [
      {
        space: {
          id: '',
          title: '3 Derry St',
          subtitle: 'Aberfeldie 3040',
          imageUrl: 'http://myImage.png',
          viewerIsMember: true,
          addressFormatted: '3 Derry St, Aberfeldie 3040',
          nextEvent: {
            name: 'Open',
            startDate: (new Date('2018-12-12')).toISOString(),
            endDate: (new Date('2018-12-12')).toISOString(),
          },
          memberships: { edges: [] },
        },
      },
    ];

    testProps.forEach((props) => {
      const wrapper = shallow(<SpaceRow {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

