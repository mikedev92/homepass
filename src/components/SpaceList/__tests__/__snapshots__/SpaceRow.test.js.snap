// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceRow /> matches snapshot 1`] = `
<Styled(ListItem)
  onPress={[Function]}
  thumbnail={true}
>
  <Styled(Left)>
    <Styled(Thumbnail)
      source={
        Object {
          "uri": "http://myImage.png",
        }
      }
      square={true}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    />
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Text)>
      3 Derry St
      
      
      <Styled(Text)
        note={true}
      >
        Aberfeldie 3040
      </Styled(Text)>
      
      
      <Styled(Text)
        note={true}
      >
        Open
         
        in a year
      </Styled(Text)>
    </Styled(Text)>
    <RowNB>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RowNB>
  </Styled(Body)>
</Styled(ListItem)>
`;
